drivers/ethernet/Kconfig.oa_tc6

```
@@ -0,0 +1,44 @@
+# Copyright (c) 2025 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig OA_TC6
+	bool "OPEN Alliance TC6 10BASE-T1x MAC-PHY support"
+	help
+	  This library implements OPEN Alliance TC6 10BASE-T1x MAC-PHY
+	  Serial Interface protocol for supporting 10BASE-T1x MAC-PHYs.
+
+if OA_TC6
+
+config OA_TC6_IRQ_THREAD_STACK_SIZE
+	int "Stack size of OA TC6 thread that processes IRQ"
+	default 512
+	help
+	  Size of the stack used for internal thread which is ran to
+	  process raised INT IRQ.
+
+config OA_TC6_IRQ_THREAD_PRIO
+	int "Priority for internal incoming packet handler"
+	default 2
+	help
+	  Priority level for internal thread which is ran for LAN
+	  INT IRQ processing.
+
+config OA_TC6_TIMEOUT
+	int "IP buffer timeout"
+	default 100
+	help
+	  Given timeout in milliseconds. Maximum amount of time
+	  that the driver will wait from the IP stack to get
+	  a memory buffer.
+
+config OA_TC6_TX_RX_BUFFER_SIZE
+	int "OA SPI TX RX buffer size (Value must be multiples of 68)"
+	default 2108
```

**2108** B - the in-tree version works with allocations of **68** B and hence allow almost any uC to use LAN865x to connect to T1S network.

---

Okay.
I will change the default value to 68.

`config OA_TC6_TX_RX_BUFFER_SIZE`
`int "OA SPI TX RX buffer size (Value must be multiples of 68)"`
`default 68`

```suggestion
	default 68
```

Is that okay?

---

@RamyaI73943 - If this change would allow the driver to still operate as supposed - i.e. this patch set shall also work on your target board when you set `OA_TC6_TX_RX_BUFFER_SIZE = 68`.
If this is the case - then yes please add such a change.

Moreover, I think that it would be also welcome to add proper information to the `Kconfig.oa_tc6` - that for SoC with more resources - the value shall be 2108. Or even better - you could just set default of 2108 for your target device' SoC family.

I shall be able to test this patchset by the end of week.

---

people can always set this value lower, if they have less resources. its just a default or the default can be set lower on board specific configs

---

> If this change would allow the driver to still operate as supposed - i.e. this patch set shall also work on your target board when you set `OA_TC6_TX_RX_BUFFER_SIZE = 68`.

@lmajewski  Yes, by setting OA_TC6_TX_RX_BUFFER_SIZE to 68, the driver works on my target board as well. In my case, the only difference is that with a value of 68, I achieve lower bandwidth, whereas setting it to 2108 allows me to achieve the maximum bandwidth of 9.46 Mbps. Since this parameter is user-configurable, users can adjust the value in menuconfig according to their available resources and requirements.


> Moreover, I think that it would be also welcome to add proper information to the `Kconfig.oa_tc6` - that for SoC with more resources - the value shall be 2108. 

Okay. I will add more information about it in help symbol. See the below changes in Kconfig.oa_tc6,
```
config OA_TC6_TX_RX_BUFFER_SIZE
	int "OASPI TX RX buffer size (Value must be multiples of 68)"
	default 68
        range 68 2108
        help
	  Size of the buffer to transmit and receive SPI chunks to MAC-PHY 
	  and from MAC-PHY respectively. In full duplex mode same buffer size
	  should be used in Tx and Rx path. Value must be multiples of 68 
          bytes (Size of the single OASPI chunk is 68 bytes).Otherwise nearest 
	  minimum multiple value of 68 will be taken. For SoCs with more
          resources, set the maximum value of 2108 to achieve higher bandwidth.
          For SoCs with limited resources, set the value to atleast 68. 

```
By adding `range 68 2108`, we specify that SoCs with more resources can increase the value up to 2108, while SoCs with limited resources should set the value to at least 68.

---

Yes, the description above provides a very detailed explanation and rationale for those values. As mentioned by @maass-hamburg you can specify `default 2108`.
I will test this patch series on my site and provide feedback.

---

Okay sure.